umbraco/Umbraco-CMS
# Unattended Install - Automated User Creation

# Feature
This feature builds upon the unattended install feature and allows the default/super user to be setup with name, email and password with configuration values. Allowing you to skip the installer and go straight to logging into the backoffice.

This works by raising an event/notification once the UnattendedInstall is complete and uses the same logic/approach as the `NewInstallStep.cs` class to modify the default super user as the installer UI would do.


## Test Notes
* Have an empty site
* Create a blank DB
* Update appsettings.development.json or use env variables to set similar config values below
* Verify the Database Schema gets created
* Goto site once its booted and login with credentials you specified in config values
* Also note with this approach without installer - no restart of dotnet run process needed it seems


## Example Settings
This PR adds three new settings that can be configured with the usual .NETCore config approach with appsettings.json files and appsettings.environment.json files and environment variables etc...


### Environment Variables
Variable Name | Example Value
------|------
ConnectionStrings__umbracoDbDSN | server=localhost;database=UmbracoUnicore;user id=sa;password='P@ssw0rd
Umbraco__CMS__Global__InstallUnattended | true
Umbraco__CMS__Global__UnattendedUserName | Warren Buckley Automation
Umbraco__CMS__Global__UnattendedUserEmail | warren@umbraco.com
Umbraco__CMS__Global__UnattendedUserPassword | password1234


### appsettings.development.json
```json
{
  "ConnectionStrings": {
    "umbracoDbDSN": "server=localhost;database=UmbracoUnicore;user id=sa;password='P@ssw0rd'"
  },
  "Umbraco": {
    "CMS": {
      "Global": {
        "InstallUnattended": true,
        "UnattendedUserName": "Warren Buckley Automation",
        "UnattendedUserEmail": "warren@umbraco.com",
        "UnattendedUserPassword": "password1234"        
      },
      "Hosting": {
        "Debug": true
      }
    }
  }
}
```
